### PR TITLE
refactor(curriculum): rewrite for RPG project

### PR DESCRIPTION
**Pull request by jdwilkin4**
## Summary

This PR is responsible for the rewrite of the RPG project descriptions in preparation of the new intro JS project that will be added before this.

For more context on the new project, here is the new project prototype

https://github.com/freeCodeCamp/CurriculumExpansion/pull/374

Naomi will be working on the new project steps and tests PR while I handle the updates to the RPG project. 

This project also simplifies the RPG project to make it easier to go through for campers. 

Here are the complete changes made in this PR:

- adding more context to steps to address the concerns of campers not understanding why changes are being made
- adding more testing of functions in these steps so campers better understand what is going on
- removing all instances of forced semicolons
- removes the easter egg steps. We have already received feedback form campers that it is hard to keep track of all that is going on. So removing this last part will simplify the project. Plus loops will be taught in the intro project so we will be fine here to remove this and shorten the project to be more manageable for campers. 

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3b79d520a7f3d0e25afd6.md
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3b79d520a7f3d0e25afd6.md:24]: wouldn't it be nice if we told the users that thet can use `.` to target classname, `#` to target id and the tag name to target tag names ? @jdwilkin4

**Review by JoyShaheb — COMMENTED** [review]
dropped some reviews @jdwilkin4

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3a75d8466a12e009eff76.md
Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3a75d8466a12e009eff76.md:12]: Yeah, we can add another sentence to provide more context on what gold and health are in the game

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3c668afc43b4a134cca81.md
Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3c668afc43b4a134cca81.md:12]: arrow functions are not taught until the music player project.
We want to keep it to the regular functions for the first three projects and then campers will learn to work with arrow functions with the music player project and onwards 👍

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3b79d520a7f3d0e25afd6.md
Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3b79d520a7f3d0e25afd6.md:24]: I vote we leave this particular sentence as is. Reason being, is because with the current version, we still have campers confused on what the syntax is for the challenge. So explicitly showing them `#example` will match what they will have to write in the challenge.

We could add another sentence, before this explaining that they could use classes, ids, tag names, etc for the querySelector.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a2509ba163e020bb9d84ea.md
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a2509ba163e020bb9d84ea.md:10]:
```suggestion
When building projects, it is best to keep your code organized and separate into HTML, CSS, and JavaScript files. 
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a2509ba163e020bb9d84ea.md
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a2509ba163e020bb9d84ea.md:12]:
```suggestion
You can use the <dfn>script</dfn> element to link your JavaScript code to your HTML file. You can then use the `src` attribute in the `script` element to specify the path to the JavaScript file that will be used to control the page.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a2509ba163e020bb9d84ea.md
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a2509ba163e020bb9d84ea.md:26]: I wouldn't have this at all since it won't be hinted at all.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a2509ba163e020bb9d84ea.md
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a2509ba163e020bb9d84ea.md:28]:
```suggestion
You should have a `script` element with a `src` attribute set to `./script.js`.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3b79d520a7f3d0e25afd6.md
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3b79d520a7f3d0e25afd6.md:10]:
```suggestion
JavaScript interacts with the HTML using the `Document Object Model (DOM)`, a tree of objects that represents the HTML. You can access the HTML by using the `document` object, which represents your entire HTML document.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3c2fccf186146b59c6e96.md
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3c2fccf186146b59c6e96.md:1]: I was able to get rid of the existing code in the seed of this step, replaced it with `button2.onclick = goCave`, and everything worked fine.

This fixed that:

```md
You should not modify the existing code adding an `onclick` attribute to `button1`.

```js
assert.match(code, /button1\.onclick\s*=\s*goStore\s*;?\s*$/m);
```
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a8a9d876b2580943ba9351.md
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a8a9d876b2580943ba9351.md:14]:
```suggestion
Objects are useful for storing data in a structured way and can represent real-world objects, like a cat.
```

**Review by Ksound22 — COMMENTED** [review]
I added a few comments.

Things look good so far.

**Comment by jdwilkin4** [comment]
Even though this PR has been reviewed in chunks when it was still in draft status, I have finished making updates to the remaining steps. So it is ready for official review.

I have added the blocked label, because this PR cannot be merged in until the pyramid project is merged in.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a7beb1ad61211ac153707f.md
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a7beb1ad61211ac153707f.md:5]: allow campers to use backticks ?

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a7bfabe119461eb13ccbd6.md
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a7bfabe119461eb13ccbd6.md:5]: allow usage of backticks ?

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a7c011eef9fb2084b966db.md
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a7c011eef9fb2084b966db.md:5]: whats the purpose of moving the function above another function ? maybe add a quick explainer or remove this mini step ?

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a8b6536156c51500739b41.md
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a8b6536156c51500739b41.md:12]:
```suggestion
Pass only the first element of the `locations` array. Remember that arrays are zero-based and the first element is at index `0`.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a8b1762b7775124622e1a3.md
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a8b1762b7775124622e1a3.md:10]: isn't it, `first the function needs to be classed` ? better remove it 

```suggestion
Instead of assigning the `innerText` and `onclick` properties to specific strings and functions, the `update` function will use data from the `location` that is passed into it.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a7c23e6b511f22ed71197a.md
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a7c23e6b511f22ed71197a.md:19]: your text
```
You should wrap the text `Store` in double quotes.
```
suggestion
```
You should wrap the text `Store` within your `text.innerText` line in double quotes.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a8a9d876b2580943ba9351.md
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a8a9d876b2580943ba9351.md:14]:
```suggestion
Objects are similar to `arrays`, except that instead of using indexes to access and modify their data, you access the data in objects through `properties`.
```

**Review by JoyShaheb — CHANGES_REQUESTED** [review]
dropped some reviews

**Comment by JoyShaheb** [comment]
hey @jdwilkin4 , is there a way to update all the tests that accepts backticks along with single and double quotes ? or let it be as it is ?

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a23c1d505bfa13747c8a9b.md
Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a23c1d505bfa13747c8a9b.md:5]: I think it would be better to open up a separate issue for this and have a discussion with the JS team on possibly making changes.  Then if the consensus is to make changes it can be handled in a separate PR. 

Even though this is a large PR with a lot of changes, I do want to minimize adding even more changes like updating portions of the seed code if we can help it. It would be better to have a separate PR for those changes so it is cleaner for reviews 👍

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a23d1c5f1c93161f3582ae.md
Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a23d1c5f1c93161f3582ae.md:5]: my earlier comment here applies 👍 

https://github.com/freeCodeCamp/freeCodeCamp/pull/53803#discussion_r1542070320

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/65f656d81d95713e0a52c927.md
Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/65f656d81d95713e0a52c927.md:23]: I was hoping to avoid adding more regex tests if possible because we have been receiving a lot of flack for the number of regex tests in the curriculum as is. 

So if there is a way to use another assert method and trim the possible space around the `monsters[fighting].name` value that would be preferable.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a7c011eef9fb2084b966db.md
Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a7c011eef9fb2084b966db.md:5]: We could add a quick note

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/65f664b393ed8547a20bafd8.md
Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/65f664b393ed8547a20bafd8.md:10]: It would be better to open a separate issue to decide on if the word console should be wrapped in backticks or if console statement should be wrapped in backticks. Or if no backticks are necessary.

Then once the JS team reaches a consensus, it can be opened up for contribution and will be cleaner to audit the project and review those changes, then in this PR.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3cfc8328d3351b95d4f61.md
Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a3cfc8328d3351b95d4f61.md:5]: It would be best to open up a separate issue to address when to use backticks in this certification. That way the team can decide on the best case moving forward with all projects.

Because some tests will look for `('"`)` while others won't. So that needs to be a separate discussion because it affects more then just this project.

**Comment by jdwilkin4** [comment]
@JoyShaheb 

> is there a way to update all the tests that accepts backticks along with single and double quotes ? or let it be as it is ?

I left a note here about that 
https://github.com/freeCodeCamp/freeCodeCamp/pull/53803#discussion_r1542087395

**Comment by jdwilkin4** [comment]
@JoyShaheb 

I will apply the remaining changes tomorrow 👍

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a8eec45f77bc69e8775294.md
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/62a8eec45f77bc69e8775294.md:14]:
```suggestion
To test your changes, click on the `"Fight dragon"` button and then click on the `"Dodge"` button. You should see the message displayed on the screen. 
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/65dc4138a809a9cbbbf3f722.md
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-dom-manipulation-by-building-a-role-playing-game/65dc4138a809a9cbbbf3f722.md:10]:
```suggestion
When you click the `"Fight dragon"` button, the `fightDragon` function should be called.
```

**Comment by Ksound22** [comment]
Other than the nitpicks I added 2 suggestions for, this is okay with me.

**Comment by jdwilkin4** [comment]
Hi @Ksound22 and @JoyShaheb ! 

Thank you for your reviews. 

As I was looking through the analytics for this project a lot of these changes for simplifying the project can be applied now and shouldn't be blocked by the release of the intro project. 

So what I am going to do is close this PR and instead break up this work into essentially two groups:

- group 1 changes: This will be one PR just for the updates needed to accommodate the intro project. So basically it will just remove the introductions to concept like let, const, functions, if/else statements, etc since that will already be covered in the intro project
- group 2 changes: This will be a series of smaller PR's to provide more instances of testing out the functions and adding more context on how things work so it is not as much of a learning curve and more beginner friendly. 

I will incorporate all of the your review feedback into those PR's 👍 

This way, we can get some of these changes out faster and it won't be blocked by the intro project 👍

**Pull request closed without merging**
